zhaquirks/philips/rwl020.py

```diff
@@ -0,0 +1,146 @@
+    TURN_OFF,
+    TURN_ON,
+)
+
+DIAGNOSTICS_CLUSTER_ID = 0x0B05  # decimal = 2821
+
+
+class BasicCluster(CustomCluster, Basic):
+    """Centralite acceleration cluster."""
+
+    attributes = Basic.attributes.copy()
+    attributes.update({0x0031: ("phillips", t.bitmap16)})
+
+
+class PhilipsRWL020(CustomDevice):
+    """Phillips RWL020 device."""
+
+    signature = {
+        #  <SimpleDescriptor endpoint=1 profile=49246 device_type=2080
+        #  device_version=2
+        #  input_clusters=[0]
+        #  output_clusters=[0, 3, 4, 6, 8]>
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zll.PROFILE_ID,
+                DEVICE_TYPE: zll.DeviceType.CONTROLLER,
+                INPUT_CLUSTERS: [Basic.cluster_id],
+                OUTPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    LevelControl.cluster_id,
+                ],
+            },
+            #  <SimpleDescriptor endpoint=2 profile=260 device_type=12
+            #  device_version=0
+            #  input_clusters=[0, 1, 3, 15, 64512]
+            #  output_clusters=[25]>
+            2: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.SIMPLE_SENSOR,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    Identify.cluster_id,
+                    BinaryInput.cluster_id,
+                    64512,
+                ],
+                OUTPUT_CLUSTERS: [Ota.cluster_id],
+            },
+        }
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                INPUT_CLUSTERS: [Basic.cluster_id],
+                OUTPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    LevelControl.cluster_id,
+                ],
+            },
+            2: {
+                INPUT_CLUSTERS: [
+                    BasicCluster,
+                    PowerConfiguration.cluster_id,
+                    Identify.cluster_id,
+                    BinaryInput.cluster_id,
+                    64512,
+                ],
+                OUTPUT_CLUSTERS: [Ota.cluster_id],
+            },
+        }
+    }
+
+    device_automation_triggers = {
+        (SHORT_PRESS, TURN_ON): {COMMAND: COMMAND_ON},
+        (SHORT_PRESS, TURN_OFF): {COMMAND: COMMAND_OFF_WITH_EFFECT},
+        (SHORT_PRESS, DIM_UP): {
+            COMMAND: COMMAND_STEP,
+            CLUSTER_ID: 8,
+            ENDPOINT_ID: 1,
+            ARGS: [0, 30, 9],
+        },
+        (LONG_PRESS, DIM_UP): {
+            COMMAND: COMMAND_STEP,
+            CLUSTER_ID: 8,
+            ENDPOINT_ID: 1,
+            ARGS: [0, 56, 9],
+        },
+        (LONG_RELEASE, "Dim Up/Down"): {
```

Comment:
These are std keys in ZHA for the translation enging... just remove the release action

Reply:
Without the (LONG_RELEASE, "Dim Up/Down"): { } block the action is not exposed to the automation engine and would have to be manually handled through zha_events.

Adding it makes it so that the automation frontend can easily assign an event to it.

Reply:
I wrote the feature support in ZHA. I know exactly what it does :)

Reply:
Doing that makes it so that the value you have supplied can’t be translated to other languages in HA. All of the keys in () actually get translated into the set language in HA. They correspond to strings in here: https://github.com/home-assistant/core/blob/d3f9408650da467ce0dbde0656837d269054566e/homeassistant/components/zha/strings.json#L25

Reply:
If both buttons (dim up and dim down) have an event the correct solution is to add 2 actions and use the appropriate type / sub_type combo for each

Reply:
I thought you meant to remove the block entirely.

None of the existing strings directly apply to the release action since there is no specific event for each button's release.  There is a generic "step" that has the same args regardless of which button is released.

What value would you prefer to have placed into the second argument of the (LONG_RELEASE, ... ) block?

Reply:
I don’t follow... you should have this block for dim up and dim down. 1 for each button and there should be a difference in the event generated?

Reply:
I agree, there *should* be a different event for the release, but alas, there is not.  Only a single "stop" event ( I misspoke in the previous comment, "stop" not "step").  I watched zha_event multiple times to confirm.  This event is the only one that fires regardless of which dim button was released:

{
    "event_type": "zha_event",
    "data": {
        "unique_id": "00:17:88:01:10:33:6e:f3:1:0x0008",
        "device_ieee": "00:17:88:01:10:33:6e:f3",
        "endpoint_id": 1,
        "cluster_id": 8,
        "command": "stop",
        "args": []
    },
    "origin": "LOCAL",
    "time_fired": "2020-03-26T16:08:00.362954+00:00",
    "context": {
        "id": "516d7dbc5a85425fb9bd7ef317ae55ee",
        "parent_id": null,
        "user_id": null
    }
}